xbmc/epg/EpgSearchFilter.cpp

```diff
@@ -207,7 +207,7 @@ int EpgSearchFilter::FilterRecordings(CFileItemList &results)
   // TODO inefficient!
   for (int iRecordingPtr = 0; iRecordingPtr < recordings.Size(); iRecordingPtr++)
   {
-    CPVRRecording *recording = recordings.Get(iRecordingPtr)->GetPVRRecordingInfoTag();
+    const CPVRRecordingPtr recording = recordings.Get(iRecordingPtr)->GetPVRRecordingInfoTag();
```

Comment:
not 100% sure on this one, how the compiler treats this for shared_ptr, but aren't you just saying that the pointer is const here, instead of the referenced data? and if you want to make this a little bit more efficient, you move the declaration of this one 2 lines up, and just assign here ;-)